src/metabase/driver.clj

```diff
@@ -940,6 +940,16 @@
   dispatch-on-initialized-driver
   :hierarchy #'hierarchy)
 
+(defmulti truncate!
+  "Delete the current contents of `table-name`.
+  If something like a SQL TRUNCATE statement is supported, we use that, but may otherwise fall back to explicitly
+  deleting rows, or dropping and recreating the table.
+  Depending on the driver, the semantics on whether triggers are fired, AUTO_INCREMENT is reset etc, is allowed to vary.
+  The application assumes that the implementation can be rolled back if inside a transaction."
```

Comment:
Are we painting ourselves into a corner here? Perhaps some databases won't support this.

Reply:
If we find a database that doesn't support `truncate` or `delete from` without rollback, we'd need a different uploads implementation anyway, either using swapping out temp tables or deleting rows using the `_mb_row_id` after inserting maybe. In both those cases we wouldn't be using a truncate function anyway, we'd use `DROP` or `DELETE FROM ... WHERE`

Reply:
Maybe we should rename the method then to indicate this? Otherwise it seems a bit easy to just implement the multi-method without thinking to check this property. Haven't got a snappy name though... `truncate-transactionally` sounds like a weird tag line.

Reply:
Yeah I was wondering that, but I figured it felt a little insignificant to be in the name. At least we could write a test to enforce that property at least for SQL-JDBC drivers. Up to you though, I'm not strongly for/against